metrics/web/lib/project_groups/presentation/widgets/project_group_dialog.dart

```diff
@@ -0,0 +1,204 @@
+
+/// A dialog that using for updating or creating project group data.
+class ProjectGroupDialog extends StatefulWidget {
+  @override
+  ProjectGroupDialogState createState() => ProjectGroupDialogState();
+}
+
+class ProjectGroupDialogState extends State<ProjectGroupDialog> {
+  /// Controls the group name text being edited.
+  final TextEditingController _groupNameController = TextEditingController();
+
+  /// Global key that uniquely identifies the [Form] widget and allows validation of the form.
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+  /// The [ChangeNotifier] that holds the project groups state.
+  ProjectGroupsNotifier _projectGroupsNotifier;
+
+  /// Controls the loading state.
+  bool _isLoading = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _projectGroupsNotifier = Provider.of<ProjectGroupsNotifier>(
+      context,
+      listen: false,
+    );
+
+    _projectGroupsNotifier.subscribeToProjectsNameFilter();
+
+    _groupNameController.text =
+        _projectGroupsNotifier.activeProjectGroupDialogViewModel?.name;
+
+    _groupNameController.addListener(() {
+      setState(() {});
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Selector<ProjectGroupsNotifier, ActiveProjectGroupDialogViewModel>(
+      selector: (_, state) => state.activeProjectGroupDialogViewModel,
+      builder: (_, activeProjectGroupDialogViewModel, ___) {
+        final createGroupButtonText = _isLoading
+            ? ProjectGroupsStrings.creatingProjectGroup
+            : ProjectGroupsStrings.createGroup;
+        final editGroupButtonText = _isLoading
+            ? ProjectGroupsStrings.savingProjectGroup
+            : ProjectGroupsStrings.saveChanges;
+
+        return Form(
+          key: _formKey,
+          child: MetricsDialog(
+            padding: const EdgeInsets.all(32.0),
+            maxWidth: 500.0,
+            title: Text(
+              activeProjectGroupDialogViewModel.id == null
+                  ? ProjectGroupsStrings.addProjectGroup
+                  : ProjectGroupsStrings.editProjectGroup,
+              style: const TextStyle(
+                fontSize: 32.0,
+                fontWeight: FontWeight.bold,
+              ),
+            ),
+            titlePadding: const EdgeInsets.symmetric(vertical: 12.0),
+            content: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                ClearableTextFormField(
+                  validator: ProjectGroupNameValidator.validate,
+                  label: ProjectGroupsStrings.nameYourGroup,
+                  controller: _groupNameController,
+                ),
+                const Padding(
+                  padding: EdgeInsets.only(top: 16.0),
+                  child: Text(ProjectGroupsStrings.chooseProjectToAdd),
+                ),
+                Container(
+                  height: 250.0,
```

Comment:
I think we should avoid setting the height here to make the popup flexible and resizable.